### ISSUE TITLE
doesn't display groups

### ISSUE THREAD

**Issue by vinny75** [dhardin/SP-PermissionManagement#11]
On SharePoint 2010, from the main screen, selecting Groups, it never gets past the fetching groups; however stepping through the code, it gets the groups and processes them just never displays them.  Selecting Users works just fine.

**Comment by vinny75** [comment]
so - guess I didn't wait long enough for the groups to load - but it takes significantly longer to load than edit users which was slow already.

**Comment by dhardin** [comment]
How slow? I've had this deployed to various site collections with low
response times. Check network tab in dev tools and see if that's bottleneck

**Comment by vinny75** [comment]
After loading the initial page, then clicking Groups, this is the only network traffic I see:
URL Protocol    Method  Result  Type    Received    Taken   Initiator   Wait‎‎  Start‎‎ Request‎‎   Response‎‎  Cache read‎‎    Gap‎‎
/_vti_bin/UserGroup.asmx    HTTPS   POST    200 text/xml    3.23 MB 15.61 s XMLHttpRequest  0   0   15  1685    13916   -750552681
/_vti_bin/UserGroup.asmx    HTTPS   POST    200 text/xml    79.32 KB    421 ms  XMLHttpRequest  0   0   15  234 172 15195

however, it's taking just shy of 3 minutes to display

**Comment by dhardin** [comment]
How many groups and users does your site collection have?

**Comment by vinny75** [comment]
660 groups; 15008 users

I used SPServices to get those numbers, didn't take long

`  $().SPServices({
      operation: "GetUserCollectionFromSite",
      async: false,
      completefunc: function(xData, Status) {
        console.log($(xData.responseXML).find("User").length)
      }
  });

  $().SPServices({
      operation: "GetGroupCollectionFromSite",
      async: false,
      completefunc: function(xData, Status) {
        console.log($(xData.responseXML).find("Group").length)
      }
  });
`

**Comment by dhardin** [comment]
Alrighty, looks like the web service is one bottleneck because it's taking
around 15 seconds to retrieve but there is a possibility that there are
zombie views that are causing the rendering to take so long. 660 groups is
not too many for a single threaded environment and should be processed
rather quickly.  I'll see if I can take a look at it today.

**Comment by dhardin** [comment]
Also there is the possibility that a group or user is corrupted. I've seen
that affect load times in the past and deleting/recreating the user/group
resolved the issue.

**Comment by vinny75** [comment]
I wouldn't think it is the web service; since 15 seconds to retrieve the data then another 2:45 to display it.  Is there a way to see which group or user it's hanging on? Turned on testing in the config but didn't seem to display anymore information.  Happy to help test, this is a pretty useful app.

**Comment by dhardin** [comment]
Yeah there's the possibility of the zombie views so it could be causing
numerous redraws and whatnot. It's been awhile since I looked at it but I
have 10 different clients using it at the moment and haven't seen the slow
render times unless it was a corrupt user (but that's typically only on the
fetching part which doesn't account for the other 2:45 minutes).

I'm gonna bring it up in a few and see but might not get time until next
week since I'm busy today and the weekend with best friend's wedding lol.

**Comment by vinny75** [comment]
No worries - great support for a free app; I get a chance this weekend, I'll try stepping through it as well to see if I can find any issues
